compiler/rustc_resolve/src/diagnostics.rs

```diff
@@ -2445,18 +2442,13 @@ enum FoundUse {
     No,
 }
 
-/// Whether a binding is part of a pattern or an expression. Used for diagnostics.
-enum IsPattern {
+/// Whether a binding is part of a pattern or a use statement. Used for diagnostics.
+enum Mode {
```

Comment:
Maybe `DiagnosticMode`? r=me after giving it a better name than just `Mode`.

Reply:
Done. But I don't have permission to merge, so I still need you to r+